accepted/future-releases/type-modifiers/feature-specification.md

```diff
@@ -447,114 +449,108 @@ class UsesAsSuperclass extends Both {}
+|`final mixin`              |No     |No     |No     |No     |No     |
+|`sealed mixin`             |No     |No     |No     |No     |**Yes**|
 
-<table><tr><td>
+The grammar is:
 
 ```
-class
-sealed class
-base class
-interface class
-final class
-abstract class
-abstract base class
-abstract interface class
-abstract final class
-```
+classDeclaration  ::= classModifiers 'class' identifier typeParameters?
+                      superclass? interfaces?
+                      '{' (metadata classMemberDeclaration)* '}'
+                      | classModifiers 'class' mixinApplicationClass
 
-</td><td>
+classModifiers    ::= 'sealed'
+                    | 'abstract'? ('base' | 'interface' | 'final')?
+                    | 'abstract'? 'base'? 'mixin'
 
-```
-mixin class
-sealed mixin class
-base mixin class
-interface mixin class
-final mixin class
-abstract mixin class
-abstract base mixin class
-abstract interface mixin class
-abstract final mixin class
-```
+mixinDeclaration  ::= mixinModifier? 'mixin' identifier typeParameters?
+                      ('on' typeNotVoidList)? interfaces?
+                      '{' (metadata classMemberDeclaration)* '}'
 
-</td><td>
-
-```
-mixin
-sealed mixin
-base mixin
-interface mixin
-final mixin
+mixinModifier     ::= 'sealed' | 'base' | 'interface' | 'final'
 ```
 
-</td></tr></table>
-
-The grammar is:
-
-```
-classDeclaration ::=
-  classModifiers 'class' identifier typeParameters?
-  superclass? interfaces?
-  '{' (metadata classMemberDeclaration)* '}'
-  | classModifiers 'class' mixinApplicationClass
-
-mixinDeclaration ::= mixinModifier? 'mixin' identifier typeParameters?
-  ('on' typeNotVoidList)? interfaces?
-  '{' (metadata classMemberDeclaration)* '}'
-
-classModifiers ::= ( 'sealed' | 'abstract'? typeModifier? ) 'mixin'?
-mixinModifier ::= 'sealed' | typeModifier
-typeModifier ::= 'base' | 'interface' | 'final'
-```
-
-### Static semantics
+## Static semantics
 
 It is a compile-time error to:
 
 *   Extend a class marked `interface` or `final` outside of the library where it
     is declared.
 
-*   Implement a type marked `base` or `final` outside of the library where it is
-    declared.
+*   Implement a class or mixin marked `base` or `final` outside of the library
+    where it is declared.
 
-*   Extend or mix in a type marked `base` outside of the library where it is
-    declared without also being marked `base` or `final`. *This ensures that a
+*   Mix in a mixin marked `interface` or `final` outside of the library where it
+    is declared.
+
+*   Extend a class marked `base` outside of the library where it is declared
+    unless the extending class is marked `base` or `final`. *This ensures that a
     subtype can't escape the `base` restriction of its supertype by offering its
     _own_ interface that could then be implemented without inheriting the
     concrete implementation from the supertype.*
 
+*   Mix in a mixin marked `base` or `final` outside of the library where it is
```

Comment:
You can't mix in a mixin marked `final` outside of the library where it is declared, right?

Reply:
Good catch, fixed.